redisraft.h

```diff
@@ -452,7 +467,8 @@ typedef struct ShardGroup {
  */
 typedef struct ShardingInfo {
     unsigned int shard_groups_num;       /* Number of shard groups */
-    ShardGroup **shard_groups;           /* Shard groups array */
+    ShardGroup **shard_groups;
+    RedisModuleDict *shard_group_map;    /* shard group id -> x in shard_groups[x] */
```

Comment:
pondering it, would make the hash_slots_map[REDIS_RAFT_HASH_SLOTS] slower.  as either it have to be dict as well.  its open for discussion

Reply:
I'm thinking we can do this, just that the hash_slots_map would point to a ShardGroup, but once a ShardGroup is allocated, it will never be freed (which sort of makes sense, it would just become a ShardGroup without any slots assigned to it), so if another ShardGroup owns the slot, we would just change the pointer its pointing to.

Reply:
@sjpotter Maybe I'm missing something here, what I had in mind was that both `shard_group_map` and `hash_slots_map` would hold pointers to `ShardGroup` structures. I don't see what is the problem to free a `ShardGroup`, as long as you also make remove it from the dict and remove its slots reference.

A slot that is not assigned to a `ShardGroup` can have a `NULL`.

Reply:
ok, that makes sense.  I think I can do that.